### PR TITLE
add logging for yum repos and compose ids

### PR DESCRIPTION
**Pull request by mlangsdorf**
Fixes OSBS-8381

Add explicit logging of injected yum repos and inherited yum repos and compose ids.

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: tests/plugins/test_resolve_composes.py [hunk 1]
```diff
@@ -447,11 +447,18 @@ def test_inherit_parents(self, workflow, reactor_config_map, parent_compose, par
         else:
             assert set(nonearch_repourls) == set(expected_yum_repourls)
 
+        if allow_inherit and parent_compose:
+            print(caplog.text)
```
Comment (athos-ribeiro) [inline tests/plugins/test_resolve_composes.py]: Could we remove this print here (this is the reason CI if failing)?

**Comment by athos-ribeiro** [comment]
Waiving docs and toggle requirements: changes to logging output only.

File: atomic_reactor/plugins/pre_resolve_composes.py [hunk 2]
```diff
@@ -168,12 +169,20 @@ def adjust_for_inherit(self):
                            build_info['nvr'], build_info['id'])
 
         all_compose_ids = set(self.compose_ids)
+        original_compose_ids = deepcopy(all_compose_ids)
         all_compose_ids.update(parent_compose_ids)
         self.all_compose_ids = list(all_compose_ids)
+        for compose_id in all_compose_ids:
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_resolve_composes.py]: Wouldn't be sufficient to list `parent_compose_ids` ?

File: atomic_reactor/plugins/pre_inject_yum_repo.py [hunk 3]
```diff
@@ -92,3 +92,5 @@ def run(self):
         yum_repos = list(self.workflow.files)
         add_yum_repos_to_dockerfile(yum_repos, df, inherited_user,
                                     self.workflow.builder.base_from_scratch)
+        for repo in yum_repos:
+            self.log.info("injected yum repo: %s", repo)
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_inject_yum_repo.py]: Do we need multi-line log entry for this?

Comment (mlangsdorf) [reply atomic_reactor/plugins/pre_inject_yum_repo.py]: I'm not sure, that's why I put it in a separate commit.

File: atomic_reactor/plugins/pre_resolve_composes.py (hunk 2 above)
Comment (mlangsdorf) [reply atomic_reactor/plugins/pre_resolve_composes.py]: The parent_compose_ids might be the same as the existing ids, in which case no new compose ids are being added.

File: atomic_reactor/plugins/pre_inject_yum_repo.py (hunk 3 above)
Comment (mlangsdorf) [reply atomic_reactor/plugins/pre_inject_yum_repo.py]: More specifically, I wasn't sure if injected repos should be logged at all, and if there were logged, whether it was better to log them one per line or as a group.  If they need to be logged, I suggest 1 per line for easier grepping.

File: tests/plugins/test_resolve_composes.py (hunk 1 above)
Comment (mlangsdorf) [reply tests/plugins/test_resolve_composes.py]: Fixed.

**Pull request merged**